source/text_handler.cpp

```diff
@@ -330,14 +330,28 @@ spv_result_t AssemblyContext::recordTypeDefinition(
     if (pInst->words.size() != 4)
       return diagnostic() << "Invalid OpTypeInt instruction";
     types_[value] = {pInst->words[2], pInst->words[3] != 0,
-                     IdTypeClass::kScalarIntegerType};
+                     IdTypeClass::kScalarIntegerType, SPV_FP_ENCODING_UNKNOWN};
   } else if (pInst->opcode == spv::Op::OpTypeFloat) {
     if ((pInst->words.size() != 3) && (pInst->words.size() != 4))
       return diagnostic() << "Invalid OpTypeFloat instruction";
-    // TODO(kpet) Do we need to record the FP Encoding here?
-    types_[value] = {pInst->words[2], false, IdTypeClass::kScalarFloatType};
+    spv_fp_encoding_t enc = SPV_FP_ENCODING_UNKNOWN;
+    if (pInst->words.size() >= 4) {
+      const char* encoding = grammar.lookupOperandName(
```

Comment:
There are new APIs (as of a month ago or so) where you don't need the grammar to do the lookup.
Also this code should not use string comparisons.

See the APIs in source/table2.h

```
spvtools::OperandDesc* desc;
if (SPV_SUCCESS == spvtools::LookupOperand( SPV_OPERAND_TYPE_FPENCODING, pInst->words[3], &desc)) {
   .. the operand is valid, use a conversion helper as suggested above.
}
```

Reply:
Done, thanks!